### PR TITLE
[MAINT] add link checker

### PR DESCRIPTION
**Pull request by Remi-Gau**

### CODE REVIEW

**Comment by Remi-Gau** [comment]
dead links:

- anything under https://schema.repronim.org/
- https://github.com/ReproNim/reproschema/blob/master/protocols/VoicePilot/VoicePilot_schema
- https://github.com/ReproNim/demo-protocol/blob/master/VoicePilot/VoicePilot_schema
- https://raw.githubusercontent.com/ReproNim/demo-protocol/master/VoicePilot/VoicePilot_schema
- templates/redcap2rs.yaml

**Comment by yibeichan** [comment]
> * anything under https://schema.repronim.org/

one of our labmates asked satra about the above, here was satra's answer: "Those are uri not URL. They aren't expected to resolve. Think of them as keys"
So we can keep them there @satra right?

> * https://github.com/ReproNim/reproschema/blob/master/protocols/VoicePilot/VoicePilot_schema

fixed in #476 

> * https://github.com/ReproNim/demo-protocol/blob/master/VoicePilot/VoicePilot_schema

fixed in #476 

> * https://raw.githubusercontent.com/ReproNim/demo-protocol/master/VoicePilot/VoicePilot_schema

fixed in #476 

> * templates/redcap2rs.yaml

fixed in #476 

cool cool, I guess we can merge this one and #476 together?

**Comment by Remi-Gau** [comment]
> > * anything under https://schema.repronim.org/
> 
> one of our labmates asked satra about the above, here was satra's answer: "Those are uri not URL. They aren't expected to resolve. Think of them as keys" So we can keep them there @satra right?

Then should I add an ignore pattern in this config file?

[mlc_config.json](https://github.com/ReproNim/reproschema/pull/467/files#diff-e23f1b945f6dffc96aa6b3cda4a7096859b6f056130bea1ddde9c60aa8989103)

**Comment by satra** [comment]
> So we can keep them there @satra right?

yes, for the moment. as a basic start we could create a schema repo and put the linkml generated documentation for the schema there. (just to make sure that we get uri's correct). in the future, we could consider turning it into a service (we had this in the past).

**Comment by yibeichan** [comment]
sounds good. @Remi-Gau can you add it to the config file to be ignored?
I'll talk to @djarecka about "create a schema repo and put the linkml generated documentation for the schema"

File: mlc_config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "ignorePatterns": [
+        {"pattern": "http://127.0.0.1:8000/"},
+        {"pattern": " https://jsonformatter.curiousconcept.com/"}
```
Comment (Remi-Gau) [inline mlc_config.json:4]:
```suggestion
        {"pattern": " https://jsonformatter.curiousconcept.com/"},
        {"pattern": " https://schema.repronim.org/"}
```

**Comment by Remi-Gau** [comment]
maybe let's merge #476 and then I can rebase this PR to make sure everything is green

**Comment by yibeichan** [comment]
done!

File: mlc_config.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+    "ignorePatterns": [
+        {"pattern": "http://127.0.0.1:8000/"},
+        {"pattern": " https://jsonformatter.curiousconcept.com/"},
+        {"pattern": " https://schema.repronim.org/.*"}
```
Comment (Remi-Gau) [inline mlc_config.json:5]:
```suggestion
        {"pattern": "^https://jsonformatter.curiousconcept.com/"},
        {"pattern": "^https://schema.repronim.org/.*"}
```

File: mlc_config.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+    "ignorePatterns": [
+        {"pattern": "http://127.0.0.1:8000/"},
+        {"pattern": "^https://jsonformatter.curiousconcept.com/"},
+        {"pattern": "^http.*://schema.repronim.org/.*"},
+        {"pattern": "^http://schema.org/version/2.0/"}
```
Comment (Remi-Gau) [inline mlc_config.json:6]: FYI this invalid link is only used as an example

https://github.com/ReproNim/reproschema/blob/f1d4c2bdee57bdc92fb911b83bbc3a2288e746d8/docs/30_schema.md?plain=1#L79

Comment (yibeichan) [reply mlc_config.json:6]: looks good! (sorry, i didn't get any notifications of this....

**Pull request merged**